builtins/amp-ad.js

```diff
@@ -284,6 +285,28 @@ export function installAd(win) {
       return loadPromise(this.iframe_);
     }
 
+    /** @override  */
+    unlayoutCallback() {
+      if (this.iframe_) {
+        removeElement(this.iframe_);
+        if (this.placeholder_) {
+          this.togglePlaceholder(true);
+        }
+        if (this.fallback_) {
+          this.toggleFallback(false);
+        }
+
+        this.iframe_ = null;
+        this.intersectionObserver_ = null;
```

Comment:
Ahh, I see that now. Both [`#pause`](https://github.com/ampproject/amphtml/blob/master/src/service/resources-impl.js#L1858) and [`#unlayout`](https://github.com/ampproject/amphtml/blob/master/src/service/resources-impl.js#L1843) call `setInViewport(false)`.